api/src/main/java/org/apache/iceberg/expressions/ResidualEvaluator.java

```diff
@@ -253,5 +286,32 @@ public Expression or(Expression leftResult, Expression rightResult) {
       // if binding didn't result in a Predicate, return the expression
       return bound;
     }
+
+    // Overriding some behaviour for inclusive projections
+    public <T> Expression predicateInclusive(BoundPredicate<T> pred) {
```

Comment:
Let me know if this could have been refactored better?

Reply:
I don't think this is needed. The `predicate` method checks whether the result of running the predicate was `alwaysTrue` for the strict projection, or `alwaysFalse` for the inclusive projection. Since you're always checking that, the predicate operation methods (`lt`, `gtEq`, `notEq`, etc.) no longer need to return `null`. That way, you can use the same predicate evaluation for both strict and inclusive projections.

Reply:
Ah, makes sense. Will make the change 👍